ts/features/pn/aar/saga/fetchNotificationDataSaga.ts

```diff
@@ -0,0 +1,156 @@
+import { ThirdPartyMessage as AarThirdPartyMessage } from "../../../../../definitions/pn/aar/ThirdPartyMessage";
+import NavigationService from "../../../../navigation/NavigationService";
+import { pnMessagingServiceIdSelector } from "../../../../store/reducers/backendStatus/remoteConfig";
+import { SessionToken } from "../../../../types/SessionToken";
+import { isTestEnv } from "../../../../utils/environment";
+import { MESSAGES_ROUTES } from "../../../messages/navigation/routes";
+import { getServiceDetails } from "../../../messages/saga/utils";
+import { profileFiscalCodeSelector } from "../../../settings/common/store/selectors";
+import PN_ROUTES from "../../navigation/routes";
+import { SendAARClient } from "../api/client";
+import {
+  EphemeralAarMessageDataActionPayload,
+  populateStoresWithEphemeralAarMessageData,
+  setAarFlowState
+} from "../store/actions";
+import { currentAARFlowData } from "../store/reducers";
+import { fillerEphemeralAARMarkdown } from "../utils/detailsById";
+import { AarFlowStates, sendAARFlowStates } from "../utils/stateUtils";
+
+export function* fetchAarDataSaga(
+  fetchData: SendAARClient["getAARNotification"],
+  sessionToken: SessionToken
+) {
+  const currentState = yield* select(currentAARFlowData);
+  if (currentState.type !== sendAARFlowStates.fetchingNotificationData) {
+    return;
+  }
+  try {
+    const result = yield* call(fetchData, {
+      Bearer: sessionToken,
+      iun: currentState.iun,
+      mandateId: currentState.mandateId
+    });
+
+    if (E.isLeft(result)) {
+      yield* put(
+        setAarFlowState({
+          type: sendAARFlowStates.ko,
+          previousState: currentState
+        })
+      );
+      return;
+    }
+    const { status, value } = result.right;
+    if (status === 200) {
+      const payload = yield* call(
+        aarMessageDataPayloadFromResponse,
+        value,
+        currentState
+      );
+
+      if (O.isNone(payload)) {
+        return;
+      }
+      const data = payload.value;
+
+      yield* put(populateStoresWithEphemeralAarMessageData(data));
+
+      yield* call(
+        NavigationService.dispatchNavigationAction,
+        StackActions.push(MESSAGES_ROUTES.MESSAGES_NAVIGATOR, {
+          screen: PN_ROUTES.MAIN,
+          params: {
+            screen: PN_ROUTES.MESSAGE_DETAILS,
+            params: {
+              messageId: data.iun,
+              serviceId: data.pnServiceID,
+              firstTimeOpening: true
+            }
+          }
+        })
+      );
+      yield* put(
+        setAarFlowState({
+          type: sendAARFlowStates.displayingNotificationData,
+          notification: value,
+          fullNameDestinatario: currentState.fullNameDestinatario,
+          mandateId: currentState.mandateId
+        })
+      );
+      return;
+    }
+    yield* put(
+      setAarFlowState({
+        type: sendAARFlowStates.ko,
+        previousState: currentState
+      })
+    );
+    return;
+  } catch (e: any) {
```

Comment:
`e` is not any, is `unknown`

PS: a better error reporting will come in a later PR